sdk/attestation/Azure.Security.Attestation/src/AttestationToken.cs

```diff
@@ -143,43 +202,390 @@ protected AttestationToken()
+                i += 1;
+            }
+            return jwkCertificates;
+        }
+
+        /// <summary>
+        /// Iterate over the set of possible signers looking for a signer which can validate the signature
+        /// </summary>
+        /// <param name="possibleSigners">A set of attestation signers which might have been used to sign the token.</param>
+        /// <returns>true if one of the possibleSigners signed the token, false otherwise</returns>
+        private bool ValidateTokenSignature(AttestationSigner[] possibleSigners)
+        {
+            bool signatureValidated = false;
+            AttestationSigner actualSigner = null;
+
+            foreach (var signer in possibleSigners)
+            {
+                // The leaf certificate is defined as the certificate which signed the token, so we just need to look
+                // at the first certificate in the chain.
+                AsymmetricAlgorithm asymmetricAlgorithm = signer.SigningCertificates[0].PublicKey.Key;
+                if (asymmetricAlgorithm is RSA rsaKey)
+                {
+                    signatureValidated = rsaKey.VerifyData(
+                        Encoding.UTF8.GetBytes(Base64Url.Encode(this.TokenHeaderBytes.ToArray()) + "." + Base64Url.Encode(this.TokenBodyBytes.ToArray())),
+                        this.TokenSignatureBytes.ToArray(),
+                        HashAlgorithmName.SHA256, RSASignaturePadding.Pkcs1);
+                    if (signatureValidated)
+                    {
+                        actualSigner = signer;
+                    }
+                    break;
+                }
+                else if (asymmetricAlgorithm is ECDsa ecdsa)
+                {
+                    signatureValidated = ecdsa.VerifyData(
+                        Encoding.UTF8.GetBytes(Base64Url.Encode(this.TokenHeaderBytes.ToArray()) + "." + Base64Url.Encode(this.TokenBodyBytes.ToArray())),
+                        this.TokenSignatureBytes.ToArray(),
+                        HashAlgorithmName.SHA256);
+                    if (signatureValidated)
+                    {
+                        actualSigner = signer;
+                    }
+                    break;
+                }
+            }
+
+            if (actualSigner != null)
+            {
+                this.CertificateThumbprint = actualSigner.SigningCertificates[0].Thumbprint;
+                this.SigningCertificate = actualSigner;
+            }
+            return actualSigner != null;
+        }
+
+        /// <summary>
+        /// Validate the common properties of the attestation token - ensure that the expiration time for the token is "reasonable", if present.
+        /// </summary>
+        /// <returns>true if the common properties are valid.</returns>
+        /// <exception cref="Exception">Thrown if the attestation token is not value.</exception>
+        private bool ValidateCommonProperties(TokenValidationOptions options)
```

Comment:
This method doesn't consider it an error if the caller has requested validation (e.g. `ValidateIssuer` == true) and the JWT is missing the corresponding field (e.g. `Payload.Issuer` == null).  This seems unexpected and wrong to me.  If the caller says to validate the issuer and the issuer isn't present, then validation should fail. 

This comment applies to all the `bool` values in the `options` class instance.

Reply:
If we adopt your suggestion, then the ValidateToken API cannot successfully validate a JWS, and one of the scenarios for this API is validating JWS's as well as JWTs. Specifically, a `StoredAttestationPolicy` is a JWS, not a JWT and this function would not be able to a JWS whose body is a `StoredAttestationPolicy`.

I've updated the documentation to reflect this.